tools/autograd/load_derivatives.py

```diff
@@ -48,34 +49,49 @@
 
 _GLOBAL_LOAD_DERIVATIVE_CACHE = {}
 
-# This function directly adds derivative entries for {view}_copy variants of each view op.
+_VALID_AUTOGRAD_KEYS = set(AUTOGRAD_KEYS)
+
+# This function directly adds per-dispatchkey derivative entries for {view}_copy variants of each view op.
 # Since every {view} and {view}_copy op shares the same derivative formula,
 # we generate them here instead of duplicating them in the yaml.
 # See Note [Codegen'd {view}_copy Operators]
 def add_view_copy_derivatives(
-    infos: List[DifferentiabilityInfo], view_groups: List[NativeFunctionsViewGroup]
-) -> List[DifferentiabilityInfo]:
+    infos: Dict[FunctionSchema, Dict[str, DifferentiabilityInfo]],
+    view_groups: List[NativeFunctionsViewGroup],
+) -> None:
     # Get the map from each view op's name to its corresponding view group
     view_name_to_group: Dict[OperatorName, NativeFunctionsViewGroup] = {
         g.view.func.name: g for g in view_groups
     }
 
-    view_copy_differentiability_infos = []
-    for info in infos:
-        maybe_view_group = view_name_to_group.get(info.func.func.name, None)
-        if maybe_view_group is not None and maybe_view_group.view_copy is not None:
-            view_copy_info = info.create_view_copy_from_view_derivative(
-                maybe_view_group
-            )
-            if view_copy_info is not None:
-                view_copy_differentiability_infos.append(view_copy_info)
+    view_infos = dict()
+
+    for _, info_dispatch_dict in infos.items():
+        # maybe_view_group only needs to be calculated once per info_dispatch_dict
+        maybe_view_group = None
```

Comment:
Why? Also since this is just a lookup, why not keep the original code?

Reply:
I think the derivatives of view_copy oops need to be generated per dispatch key, which is why I modified the code is my understanding correct here?

Reply:
Sure, but you could have kept the `maybe_view_group = view_name_to_group.get(info.func.func.name, None)` call inside the inner for loop.